swap_meet/item.py

```
@@ -1,2 +1,26 @@
+#Wave 2 , 5
+import uuid
+#from swap_meet.vendor import Vendor
 class Item:
-    pass
\ No newline at end of file
+    def __init__(self,id = None,category = "", condition = 0 ):
+        if id is None :
+            self.id = uuid.uuid4().int
+        else:
+            self.id = id
```

Nice solution, another option here could use conditional assignment:
```python
# Assigns self.id to the parameter id if id is truthy, 
# otherwise assigns self.id the return value of uuid.uuid4().int
self.id = id if id else uuid.uuid4().int
```

---

I was wondering the how would I raise value error if ID is not integer with this change? 
Should I add this after the above statement 
if not isinstance(id,int):
      raise ValueError("Id must be an interger")

---

Hmm, are you thinking about if the user passes in a non-integer `id` argument? In that case I'd likely add a guard statement at the top that checks if `id` is not `None` and the type of `id` is not an `int` - in that case we would raise an error. If we get past that statement then either the `id` argument was passed and has an int value, or it is `None` and we can create our default with `uuid`.

---

Yes, I want to make sure if the id entered by user is not an integer it should raise an error. Yes the guard statement will be good at the beginning so if the id is not an int we don't need to assign the id to that instance object. Will make the changes. Thank you.